[macOS] TestWebKitAPI.ServiceWorkers API test failures

salinas-miguel opened the pull request:
#### 76649060e74759fd21a4c39cfc30883e21835271
<pre>
[macOS] TestWebKitAPI.ServiceWorkers API test failures
<a href="https://bugs.webkit.org/show_bug.cgi?id=251032">https://bugs.webkit.org/show_bug.cgi?id=251032</a>
rdar://104436315

Reviewed by Youenn Fablet.

Prevent ProcessThrottler from changing WebProcessProxy&apos;s throttle state
when testing WebProcessProxy&apos;s throttle state.

* Source/WebKit/UIProcess/WebProcessProxy.cpp:
(WebKit::WebProcessProxy::setThrottleStateForTesting):
(WebKit::WebProcessProxy::didChangeThrottleState):
* Source/WebKit/UIProcess/WebProcessProxy.h:
(WebKit::WebProcessProxy::setThrottleStateForTesting): Deleted.

Canonical link: <a href="https://commits.webkit.org/259303@main">https://commits.webkit.org/259303@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/7388e3904d8c6795105ee72434aff5fd551c0254

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/104430 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/13509 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/37338 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/113646 "Built successfully") | 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/14612 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/4425 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/96655 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/112683 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/110198 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/11262 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/94322 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/38878 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/93127 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/25933 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/80547 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/6868 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/27290 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/6997 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/3843 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/13025 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/46848 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/6413 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/8791 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

salinas-miguel commented on Source/WebKit/UIProcess/WebProcessProxy.cpp:
```diff
@@ -1561,8 +1561,17 @@ void WebProcessProxy::sendProcessDidResume(ResumeReason)
+{
+    m_areThrottleStateChangesEnabled = true;
+    didChangeThrottleState(state);
+    m_areThrottleStateChangesEnabled = false;
```
This wasn't needed before https://github.com/WebKit/WebKit/pull/5759 because `ProcessThrottler` would rarely fire `didChangeThrottleState()` on macOS. Now that it fires frequently, it overwrites the state after tests call `setThrottleStateForTesting()`.

youennf reviewed (approved):
Seems ok.
An alternative would be to actually make the web view go to background (by minimizing them for instance), maybe play audio and so on, then validate what process assertions are taken for the service worker process.

salinas-miguel committed:
[macOS] TestWebKitAPI.ServiceWorkers API test failures
https://bugs.webkit.org/show_bug.cgi?id=251032
rdar://104436315

Reviewed by Youenn Fablet.

Prevent ProcessThrottler from changing WebProcessProxy's throttle state
when testing WebProcessProxy's throttle state.

* Source/WebKit/UIProcess/WebProcessProxy.cpp:
(WebKit::WebProcessProxy::setThrottleStateForTesting):
(WebKit::WebProcessProxy::didChangeThrottleState):
* Source/WebKit/UIProcess/WebProcessProxy.h:
(WebKit::WebProcessProxy::setThrottleStateForTesting): Deleted.

Canonical link: https://commits.webkit.org/259303@main
